src/config/config-initialiser.js

```diff
@@ -166,14 +167,54 @@ function handlePlugins (config) {
     if (plugin) {
       const PluginConstructor = resolvePluginClass(plugin, typeMap[connectorMap[key]])
       config[key] = new PluginConstructor(plugin.options)
-      if(config.pluginTypes.indexOf(key) === -1) {
+      if (config.pluginTypes.indexOf(key) === -1) {
         config.pluginTypes.push(key)
       }
     }
   }
 }
 
 /**
+ *
+ */
+function handleConnectionEndpoints (config) {
+  if (!config.connectionEndpoints || Object.keys(config.connectionEndpoints).length === 0) {
+    throw new Error('No connection endpoints configured')
+  }
+  if (Object.keys(config.connectionEndpoints).length > 1) {
+    throw new Error('Currently only one connection endpoint may be configured')
+  }
+  const connectionEndpoints = []
+  for (const connectionType in config.connectionEndpoints) {
```

Comment:
cant we now do something like

const [ connectionType, plugin ] in config.connectionEndpoints?

Don't need to change, just curious